### PR TITLE
fix(hogql): Fix retention event time in period

### PR DESCRIPTION
**Pull request by webjunkie**
## Problem

Both legacy and HogQL queries apply "to start of period" to the event time and only then compare it with the specific date of the whole query.

For monthly retention for example:

- e.g. periods start at the 15th of the month
- an event e.g. at `2023-11-20` would be set to `2023-11-01` 
- the comparison was `2023-11-01 >= 2023-11-15` so the events in all of this month were not included

Not sure why this didn't have implications for weekly as well 🤔 

## Changes

- fix so we compare to the start of the interval
- `2023-11-01 >= 2023-11-01` ✔️ 
- this means that we always had monthly cohorts starting at the 1st of the month
- fixed labels to reflect that
- did not adjust legacy query -> live compare shows old 0s or other differences now

image

## How did you test this code?

- choose monthly retention
- play with number of months
- see that previously the first cohort was 0
- now fixed
- checked tests and snapshots

### CODE REVIEW

File: frontend/src/scenes/retention/retentionTableLogic.ts [hunk 1]
```diff
@@ -73,28 +73,42 @@ export const retentionTableLogic = kea<retentionTableLogicType>([
...
+                    } else {
+                        switch (period) {
+                            case 'Hour':
+                                firstColumn = dayjs(currentResult.date).format('MMM D, h A')
+                                break
+                            case 'Month':
+                                firstColumn = dayjs(currentResult.date).format('MMM YYYY')
+                                break
+                            case 'Week':
+                                firstColumn = `Week of ${dayjs(currentResult.date).format('MMM D')}`
```
Comment (mariusandra) [inline frontend/src/scenes/retention/retentionTableLogic.ts:91]: The small problem with this is if it's "Week of [day that's Sunday]", Americans will believe that's when the week starts, but Europeans will assume that's when the week ends. So maybe "week starting..." makes it even clearer?

**Review by mariusandra — APPROVED** [review]
Nice!

File: frontend/src/scenes/retention/retentionTableLogic.ts (hunk 1 above)
Comment (webjunkie) [reply frontend/src/scenes/retention/retentionTableLogic.ts:91]: Makes sense. But we do take into account the week start day setting here or don't we? I read about it but haven't tested.

Comment (mariusandra) [reply frontend/src/scenes/retention/retentionTableLogic.ts:91]: I'm.... not too sure 😅. Probably not. But if not, we should at _some_ point...

Comment (webjunkie) [reply frontend/src/scenes/retention/retentionTableLogic.ts]: It works for me... can change from Monday to Sunday 🎉

**Pull request merged**